app/views/datasets/_form.html.erb

```diff
@@ -21,39 +35,103 @@
         <% end %>
 
         <%= form.hidden_field :work_id %>
+        <%= render 'titles_form', dataset: @dataset %>
+      </div>
 
-        <div class="field">
-          <%= form.label :title %><br>
-          <%= form.text_field :title %>
-        </div>
+      <div class="tab-pane fade" id="v-pills-creator" role="tabpanel" aria-labelledby="v-pills-creator-tab">
+        <%= render 'creators_form', dataset: @dataset %>
+      </div>
 
-        <div class="field">
-          <%= form.label :collection %><br>
-          <%= form.collection_select :collection_id, current_user.submitter_collections, :id, :title, {:include_blank => true}, {:class => 'form-control'} %>
-        </div>
+      <div class="tab-pane fade" id="v-pills-identifiers" role="tabpanel" aria-labelledby="v-pills-identifiers-tab">
+        identifiers go here
+      </div>
 
-        <div class="field">
-          ARK: <%= form.object.ark %>
-        </div>
-        <br />
-
-        <div class="actions">
-          <%= form.submit class: "btn btn-primary" %>
-          <%= link_to 'Cancel', @dataset, class: "btn btn-secondary" %>
-        </div>
-      <% end %>
-    </div>
-    <div class="tab-pane fade" id="v-pills-creator" role="tabpanel" aria-labelledby="v-pills-creator-tab">
-      creators go here
-    </div>
-    <div class="tab-pane fade" id="v-pills-identifiers" role="tabpanel" aria-labelledby="v-pills-identifiers-tab">
-      identifiers go here
+      <div class="tab-pane fade" id="v-pills-publisher" role="tabpanel" aria-labelledby="v-pills-publisher-tab">
+        publisher goes here
+      </div>
+
+      <div class="tab-pane fade" id="v-pills-type" role="tabpanel" aria-labelledby="v-pills-type-tab">
+        resource type goes here
+      </div>
     </div>
-    <div class="tab-pane fade" id="v-pills-publisher" role="tabpanel" aria-labelledby="v-pills-publisher-tab">
-      publisher goes here
+
+    <div class="actions">
+      <%= form.submit class: "btn btn-primary" %>
+      <%= link_to 'Cancel', @dataset, class: "btn btn-secondary" %>
     </div>
-    <div class="tab-pane fade" id="v-pills-type" role="tabpanel" aria-labelledby="v-pills-type-tab">
-      resource type goes here
+
+    <!--
+      We use these hidden fields to keep track of how many creators and titles are on the form
+      initially and how many are added on the fly. This helps to process the data when the form
+      is submitted.
+    -->
+    <div>
+      <input type="text" id="existing_title_count" name="existing_title_count" value="<%= @datacite.titles.count %>" class="hidden" />
+      <input type="text" id="new_title_count" name="new_title_count" value="0" class="hidden" />
+      <input type="text" id="existing_creator_count" name="existing_creator_count" value="<%= @datacite.creators.count %>" class="hidden" />
+      <input type="text" id="new_creator_count" name="new_creator_count" value="0" class="hidden" />
```

Comment:
`existing_title_count` and `new_title_count` are used to track how many titles were on the form and how many the user added. This way when the user clicks Save the logic in the controller is pretty simple to parse the data that comes in the HTML form. The `existing_creator_count` and `new_creator_count` are used the same way but to track creators.